### ISSUE TITLE
Question about result reproduction

### ISSUE THREAD

**Issue by Lucky0775** [FuyaLuo/PearlGAN#3]
Hi, thanks for your nice work！

I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.

Can you further specify what needs to be done to get close to the results in the paper?

And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?

Best Regards,

**Comment by FuyaLuo** [comment]
> Hi, thanks for your nice work！
> 
> I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.
> 
> Can you further specify what needs to be done to get close to the results in the paper?
> 
> And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?
> 
> Best Regards,

Flawlessly reproducing the results is extremely difficult for image-to-image (I2I) translation models. Unlike CNN models, the training of I2I translation models is generally unsupervised, and the encoders of both domains may be switched randomly at each iteration. In addition, the random cropping of batchsize to 1 and the sensitivity of the discriminator to samples constrain the stability of training. Moreover, there are other factors such as dropout that can have an impact on GAN model training. However, under reasonable initialization conditions, a similar result can usually be obtained by multiple attempts.

In the proposed method, since the attentional loss is unsupervised, which may further affect the stability of the training, we suggest observing whether the attention maps have a hierarchical distribution along the height at around the 30th epoch of training.

For the pre-processing of the KAIST dataset, please remember to enhance the DC images as described in the paper.

**Comment by Lucky0775** [comment]
> > Hi, thanks for your nice work！
> > I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.
> > Can you further specify what needs to be done to get close to the results in the paper?
> > And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?
> > Best Regards,
> 
> Flawlessly reproducing the results is extremely difficult for image-to-image (I2I) translation models. Unlike CNN models, the training of I2I translation models is generally unsupervised, and the encoders of both domains may be switched randomly at each iteration. In addition, the random cropping of batchsize to 1 and the sensitivity of the discriminator to samples constrain the stability of training. Moreover, there are other factors such as dropout that can have an impact on GAN model training. However, under reasonable initialization conditions, a similar result can usually be obtained by multiple attempts.
> 
> In the proposed method, since the attentional loss is unsupervised, which may further affect the stability of the training, we suggest observing whether the attention maps have a hierarchical distribution along the height at around the 30th epoch of training.
> 
> For the pre-processing of the KAIST dataset, please remember to enhance the DC images as described in the paper.

Thank you for your reply!

I have augmented the DC images as described in the paper, then resized and center cropped all images using **transforms.Resize()** and **transforms.CenterCrop()** respectively to form my own dataset and finally retrained the model .

The **semantic segmentation result (mIoU_9)** of both my retraining model and the provided pretrained model on the KAIST dataset is only about 34, which is far from the 43.1 reported in the file of **Complete semantic segmentation results.md**.

Could this result be caused by the wrong operation of data-preprocessing in the process of making the dataset?

I would love to study this paper and reproduce its results. If it is convenient for you, can you share the relevant datasets used in this article (ie preprocessed kaist dataset and flir dataset) to help me reproduce the results?

Sincerely,

**Comment by FuyaLuo** [comment]
> > > Hi, thanks for your nice work！
> > > I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.
> > > Can you further specify what needs to be done to get close to the results in the paper?
> > > And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?
> > > Best Regards,
> > 
> > 
> > Flawlessly reproducing the results is extremely difficult for image-to-image (I2I) translation models. Unlike CNN models, the training of I2I translation models is generally unsupervised, and the encoders of both domains may be switched randomly at each iteration. In addition, the random cropping of batchsize to 1 and the sensitivity of the discriminator to samples constrain the stability of training. Moreover, there are other factors such as dropout that can have an impact on GAN model training. However, under reasonable initialization conditions, a similar result can usually be obtained by multiple attempts.
> > In the proposed method, since the attentional loss is unsupervised, which may further affect the stability of the training, we suggest observing whether the attention maps have a hierarchical distribution along the height at around the 30th epoch of training.
> > For the pre-processing of the KAIST dataset, please remember to enhance the DC images as described in the paper.
> 
> Thank you for your reply!
> 
> I have augmented the DC images as described in the paper, then resized and center cropped all images using **transforms.Resize()** and **transforms.CenterCrop()** respectively to form my own dataset and finally retrained the model .
> 
> The **semantic segmentation result (mIoU_9)** of both my retraining model and the provided pretrained model on the KAIST dataset is only about 34, which is far from the 43.1 reported in the file of **Complete semantic segmentation results.md**.
> 
> Could this result be caused by the wrong operation of data-preprocessing in the process of making the dataset?
> 
> I would love to study this paper and reproduce its results. If it is convenient for you, can you share the relevant datasets used in this article (ie preprocessed kaist dataset and flir dataset) to help me reproduce the results?
> 
> Sincerely,

Did you compare the sampled and processed NTIR test images with the semantic annotation? Our pre-processing of the image is performed by MATLAB, and the semantic segmentation results are shown in the image.
![res_PearlGAN_KAIST](https://user-images.githubusercontent.com/26736293/179687647-44d09213-2c83-4869-bbbf-8ec56e46cf7b.PNG)

**Comment by FuyaLuo** [comment]
> > > Hi, thanks for your nice work！
> > > I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.
> > > Can you further specify what needs to be done to get close to the results in the paper?
> > > And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?
> > > Best Regards,
> > 
> > 
> > Flawlessly reproducing the results is extremely difficult for image-to-image (I2I) translation models. Unlike CNN models, the training of I2I translation models is generally unsupervised, and the encoders of both domains may be switched randomly at each iteration. In addition, the random cropping of batchsize to 1 and the sensitivity of the discriminator to samples constrain the stability of training. Moreover, there are other factors such as dropout that can have an impact on GAN model training. However, under reasonable initialization conditions, a similar result can usually be obtained by multiple attempts.
> > In the proposed method, since the attentional loss is unsupervised, which may further affect the stability of the training, we suggest observing whether the attention maps have a hierarchical distribution along the height at around the 30th epoch of training.
> > For the pre-processing of the KAIST dataset, please remember to enhance the DC images as described in the paper.
> 
> Thank you for your reply!
> 
> I have augmented the DC images as described in the paper, then resized and center cropped all images using **transforms.Resize()** and **transforms.CenterCrop()** respectively to form my own dataset and finally retrained the model .
> 
> The **semantic segmentation result (mIoU_9)** of both my retraining model and the provided pretrained model on the KAIST dataset is only about 34, which is far from the 43.1 reported in the file of **Complete semantic segmentation results.md**.
> 
> Could this result be caused by the wrong operation of data-preprocessing in the process of making the dataset?
> 
> I would love to study this paper and reproduce its results. If it is convenient for you, can you share the relevant datasets used in this article (ie preprocessed kaist dataset and flir dataset) to help me reproduce the results?
> 
> Sincerely,

If the dataset of your processed segmentation task does not match the annotation, we suggest you test the mAP results of the KAIST dataset using the trained weights. First, resize the original image to 360*288 for NTIR2DC translation, and then upsample the DC image to 640*512 using MATLAB. Finally, the mAP results of pedestrian detection are tested using YOLOv4.

**Comment by FuyaLuo** [comment]
> > > Hi, thanks for your nice work！
> > > I aming to reproduce the results (mIoU) of semantic segmentation on the kaist dataset with the pretrained model and the re-trained version, but the results are not matched to those reported in the paper and the file of Complete semantic segmentation results.md.
> > > Can you further specify what needs to be done to get close to the results in the paper?
> > > And,are there any other steps that need to be paid attention to in the process of data-preprocessing to form the Datasets and the detection of mIoU?
> > > Best Regards,
> > 
> > 
> > Flawlessly reproducing the results is extremely difficult for image-to-image (I2I) translation models. Unlike CNN models, the training of I2I translation models is generally unsupervised, and the encoders of both domains may be switched randomly at each iteration. In addition, the random cropping of batchsize to 1 and the sensitivity of the discriminator to samples constrain the stability of training. Moreover, there are other factors such as dropout that can have an impact on GAN model training. However, under reasonable initialization conditions, a similar result can usually be obtained by multiple attempts.
> > In the proposed method, since the attentional loss is unsupervised, which may further affect the stability of the training, we suggest observing whether the attention maps have a hierarchical distribution along the height at around the 30th epoch of training.
> > For the pre-processing of the KAIST dataset, please remember to enhance the DC images as described in the paper.
> 
> Thank you for your reply!
> 
> I have augmented the DC images as described in the paper, then resized and center cropped all images using **transforms.Resize()** and **transforms.CenterCrop()** respectively to form my own dataset and finally retrained the model .
> 
> The **semantic segmentation result (mIoU_9)** of both my retraining model and the provided pretrained model on the KAIST dataset is only about 34, which is far from the 43.1 reported in the file of **Complete semantic segmentation results.md**.
> 
> Could this result be caused by the wrong operation of data-preprocessing in the process of making the dataset?
> 
> I would love to study this paper and reproduce its results. If it is convenient for you, can you share the relevant datasets used in this article (ie preprocessed kaist dataset and flir dataset) to help me reproduce the results?
> 
> Sincerely,

If the dataset of your processed segmentation task does not match the annotation, we suggest you test the mAP results of the KAIST dataset using the trained weights. First, resize the original image to 360x288 for NTIR2DC translation, and then upsample the DC image to 640x512 using MATLAB. Finally, the mAP results of pedestrian detection are tested using YOLOv4.

**Comment by ghost** [comment]
我最近开始用kaist数据集，也遇到了这个问题，
![image](https://github.com/FuyaLuo/PearlGAN/assets/38597848/2a80736b-1761-49dc-a06a-55a4be11dc50)
我对原始图片这样操作后，用预训练的权重跑了预测结果对不上，这是可视化的对比，
![set4_1_I00000](https://github.com/FuyaLuo/PearlGAN/assets/38597848/24ed20bf-9b5a-46ee-9b9a-dedc2c28f843)
![set4_1_I00000_label](https://github.com/FuyaLuo/PearlGAN/assets/38597848/be974a4f-c72d-4435-b856-5c9db528e88a)

感觉是不是kaist数据集和flir数据集预处理时候的放缩尺寸并不都是（500,400）？因为我之前在flir数据集上面这样预处理结果是对得上的。下面是flir的可视化，就通过天空区域可以明显看出flir的是匹配的，但是kaist的就不匹配（预训练权重算的数据也是flir是匹配的，但是kaist就不匹配）
![FLIR_08863](https://github.com/FuyaLuo/PearlGAN/assets/38597848/60c655d4-596e-4c9a-8b28-7e7d1cfbb89b)
![FLIR_08863_label](https://github.com/FuyaLuo/PearlGAN/assets/38597848/580b69c7-e2fc-4670-bb0f-bb77872b7aae)
不知道帅气的作者大大能不能提供下做预处理的pipeline，十分感谢！！！

**Comment by ghost** [comment]
早上来试了一下，kaist的数据集预处理应该是没有resize到【500,400】这一步的，只需要centercrop就够了，然后结果也和论文对上了
![image](https://github.com/FuyaLuo/PearlGAN/assets/38597848/918cae4a-a608-4b5f-a1ff-3025f2b1df69)

**Comment by FuyaLuo** [comment]
> 早上来试了一下，kaist的数据集预处理应该是没有resize到【500,400】这一步的，只需要centercrop就够了，然后结果也和论文对上了 ![image](https://private-user-images.githubusercontent.com/38597848/296913703-918cae4a-a608-4b5f-a1ff-3025f2b1df69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYzNDM2NzYsIm5iZiI6MTcwNjM0MzM3NiwicGF0aCI6Ii8zODU5Nzg0OC8yOTY5MTM3MDMtOTE4Y2FlNGEtYTYwOC00YjVmLWExZmYtMzAyNWYyYjFkZjY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI3VDA4MTYxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhZjc1YTljZjk4MTVmNzJlMWM1NTI5M2UwNzBmMmQxODVlMjRiYjlkZWM4NjljM2M3YzQ5Y2E0Y2NjNjgxY2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.A0JCQPeg7d7sjnIPRl_0UqlWCuoaTSQWifQHgi6fdKU)

感谢你的认真核对，能复现就好。KAIST数据集的处理时间较早，当时可能是我个人疏忽检查，导致resize到500x400这一行代码没有取消注释，我很抱歉。我后续会在README里面更正一下KAIST数据集实验的预处理步骤，以及致谢中感谢你的勘误。感谢你的核对！

**Comment by liting1018** [comment]
> transforms.Resize

请问你贴的这行代码是自己对原始数据集（640*512）进行预处理的代码嘛，我好像没有看到作者大大开源的代码里有这行。
